2_process.R

```diff
@@ -15,6 +16,28 @@ p2_targets_list <- list(
     subset_closest_nhd(nhd_lines = p1_nhd_reaches_along_NHM,
                        sites = p1_drb_temp_sites_sf)
   ),
+
+  tar_target(p2_buffered_nhd_reaches_along_nhm,
+              st_buffer(p1_nhd_reaches_along_NHM, dist = units::set_units(250, m))
+  ),
+  
+  # Reach -- depth_to_bedrock data for each nhm reach buffered 250m  
+  ## Note: If you do not have Shangguan_dtb_cm_250m_clip_path data, you must grab it from caldera project folder. 
+  ## Dataset accessible on caldera in project folder sub-dir: 1_fetch/in. scp to to local in the 1_fetch/in folder in order to run this piece of pipeline
+  ## original source: http://globalchange.bnu.edu.cn/research/dtb.jsp. Data was clipped to drb before getting added to caldera.
+  tar_target(p2_depth_to_bedrock_reaches_along_nhm,
```

Comment:
`p2_depth_to_bedrock_catchments_along_nhm_dissolved` below contains 458 rows representing the depth to bedrock values for each NHM/PRMS segment (minus `287_1`) aggregated for the catchment.

For `p2_depth_to_bedrock_reaches_along_nhm`, we similarly want 458 (or 459) rows that each correspond to a NHM/PRMS segment but aggregated for that segment's buffered reach instead of the full catchment. Currently, `p2_depth_to_bedrock_reaches_along_nhm` has 3,229 rows representing all of the COMIDs that overlap the NHM segments. So I think we either need to fetch the NHM segments from the temperature data release to use in this target, or we need to join `p1_nhd_reaches_along_NHM` with `p1_drb_comids_all_tribs` like you did in `p1_nhm_catchments_dissolved` and then dissolve the buffers so that we have one buffered reach per NHM/PRMS segid.   Does that make sense?

Reply:
I'm not as familiar with this raster dataset so I haven't inspected the output very closely, but my latter suggestion in the comment above would look something like this:

```r
# not sure what the most informative name for this target would be; 
# target includes buffered area for NHM/PRMS reaches based on the combined NHDPlusv2 reaches.
  tar_target(p2_buffered_nhd_reaches_along_nhm,
             left_join(p1_nhd_reaches_along_NHM %>% mutate(COMID = as.character(comid)),
                       p1_drb_comids_all_tribs %>% mutate(COMID = as.character(COMID)),
                       by = 'COMID') %>%
               group_by(PRMS_segid) %>%
               dplyr::summarize(geometry = sf::st_union(geometry)) %>%
               sf::st_buffer(dist = units::set_units(250, m))
  ),
```

Does that work? Or are there holes/other processing artifacts that suggest we should fetch the NHM flowlines instead?

Reply:
No problem, thanks for testing that!

Reply:
> 2/ It might be easier to reaches to PRMS scale before dissolving in this case. I ran the above and I getting the expected output, but not able to visualize. I have not spent much time debugging that.

OK, do you think it makes sense then to just download the NHM/PRMS flowlines from the temperature data release, and then do the buffer step on those?

Reply:
I don't think we would use the 158 buffer target for anything else, so I'd probably opt to include the code that creates the buffers in a separate function, assuming we plan on keeping this approach rather than buffering the NHM flowlines directly.

Reply:
> assuming we plan on keeping this approach rather than buffering the NHM flowlines directly.

Related to this, does the process that created the figure above output 1 polygon (i.e., one buffered polygon that corresponds with the area surrounding segment 158_1), or does it output 8+ polygons, each representing a separate COMID belonging to that NHM/PRMS segment? I think we want the former, since we're aiming for one value for area-weighted average depth to bedrock for the buffered segment.